ts/screens/services/ServiceDetailsScreen.tsx

```diff
@@ -100,155 +95,134 @@ const contextualHelpMarkdown: ContextualHelpPropsMarkdown = {
  * Screen displaying the details of a selected service. The user
  * can enable/disable the service and customize the notification settings.
  */
-class ServiceDetailsScreen extends React.Component<Props, State> {
-  get serviceId() {
-    return this.props.navigation.getParam("service").service_id;
-  }
+const ServiceDetailsScreen = (props: Props) => {
+  const serviceId = props.navigation.getParam("service").service_id;
+  const [isMarkdownLoaded, setIsMarkdownLoaded] = useState(false);
 
-  constructor(props: Props) {
-    super(props);
-    // We initialize the UI by making the states of the channels the same
-    // as what is set in the profile. The user will be able to change the state
-    // via the UI and the profile will be updated in the background accordingly
-    this.state = {
-      uiEnabledChannels: getEnabledChannelsForService(
-        this.props.profile,
-        this.serviceId
-      ),
-      isMarkdownLoaded: false
-    };
-  }
+  useEffect(() => {
+    props.loadServiceDetail(serviceId);
+  }, []);
 
-  public componentDidUpdate(prevProps: Props) {
-    if (pot.isError(this.props.profile) && !pot.isError(prevProps.profile)) {
-      // in case of new or resolved errors while updating the profile, we show a toast and
-      // reset the UI to match the state of the profile preferences
-      showToast(
-        I18n.t("serviceDetail.onUpdateEnabledChannelsFailure"),
-        "danger"
-      );
+  const onMarkdownEnd = () => setIsMarkdownLoaded(true);
 
-      const uiEnabledChannels = getEnabledChannelsForService(
-        this.props.profile,
-        this.props.navigation.getParam("service").service_id
-      );
-      this.setState({
-        uiEnabledChannels
-      });
-    }
-  }
-  // collect the service
-  private service = this.props.navigation.getParam("service");
-
-  private onMarkdownEnd = () => {
-    this.setState({ isMarkdownLoaded: true });
-  };
+  const { potService } = props;
 
-  public render() {
-    const { service } = this;
-
-    const metadata = service.service_metadata;
+  // This has been considered just to avoid compiling errors
+  // once we navigate from list or a message we always have the service data since they're previously loaded
+  if (potService === undefined) {
+    return null;
+  }
+  const maybeService = pot.toOption(potService);
 
-    // if markdown content is not available, render immediately what is possible
-    // but we must wait for metadata load to be completed to avoid flashes
-    const isMarkdownAvailable = metadata?.description;
-    const isMarkdownLoaded = isMarkdownAvailable
-      ? this.state.isMarkdownLoaded
-      : true;
-    // if markdown data is available, wait for it to be rendered
-    const canRenderItems = isMarkdownLoaded;
+  if (maybeService.isNone()) {
+    return null;
+  }
```

Comment:
I see that `potService` is only used to extract a nullable `service`.
You can make this part simpler by delegating the transformation to `mapStateToProps`:

```
const mapStateToProps = (state: GlobalState, props: OwnProps) => {
    // ....
   return {
      service: fromNullable(serviceByIdSelector(serviceId)(state)).chain(pot.toOption).toUndefined()
      // ...
   }
```

And then use it from `props`:

```typescript
const ServiceDetailsScreen = ({service}: Props) => {
   // exit condition first, clearly visibile
   if (service === undefined) {
      return null
   }
   // service is available here
```

Reply:
Nice suggestion, looks shorter indeed